static/shared/js/poll_data.js

```diff
@@ -103,6 +103,11 @@ export class PollData {
                 // All message readers may add a new option to the poll.
                 const idx = data.idx;
                 const option = data.option;
+                const options = this.get_widget_data().options;
+
+                if (this.is_option_present(options, option)) {
+                    return;
+                }
```

Comment:
This deserves a comment explaining that the UI won't allow users to add duplicate options, and this is primarily here because when creating a poll in the initial message, it is technically possible to have duplicates.